### DISCUSSION TITLE
Get renderer to force update

### DISCUSSION THREAD

**Discussion by jdiazgon55** [google/model-viewer#2592] [Q&A] [answered]
Hi guys,

I would like to access the `renderer` used by model-viewer in order to force an update (`renderer.render(...)`). I have been able to retrieve the `scene` and even the `SmoothControls` with:

```js
modelViewer.addEventListener(
  "load",
  (event) => {
    sceneObj = Object.getOwnPropertySymbols(modelViewer).find(
      (x) => x.description === "scene"
    );

    controls = Object.getOwnPropertySymbols(modelViewer).find(
      (x) => x.description === "controls"
    );

    controls = modelViewer[controls];

    scene = modelViewer[sceneObj];
)
```

But I am not able to get the `renderer`. I do need it because I am adding `TransformControls` to the scene and I need to disable the camera rotation while translating an object:

![image](https://user-images.githubusercontent.com/28304354/126194823-3b056012-5df3-458f-9490-b97fc37f92d1.png)

The problem is that when I disable interaction, no frame is being rendered.

```js
transformControls.addEventListener("dragging-changed", function (event) {
      if (controls.interactionEnabled) {
        controls.disableInteraction();
      } else {
        controls.enableInteraction();
      }
    });
```

I would like to `render` when the object is being moved:

```js
transformControls.addEventListener("change", () => {renderer.render(...)});
```

**Comment by elalish** [comment, marked as answer]
Well, hopefully you already know this, but for anyone else reading this, Symbols are our way of denoting private members within the non-private nature of JS. Therefore this is outside of our public API and I make no promises about stability. However, feel free to enjoy hacking! I think you might like to call `modelviewer[$needsRender]()`.

**Reply by jdiazgon55** [reply]
Thanks for the quick response! That lead me to the correct solution.

I'm not sure why but I was not able to retrieve the variable `$needsRender`, but in the source code I found out that it calls `this[$scene].isDirty = true;`. Therefore, as I already had the `scene` object, I implemented the following:

```js
    transformControls.addEventListener("change", () => {
      scene.isDirty = true;
    });
```

That works like a charm, so thanks for the help :)
